ntoskrnl/include/internal/ps_i.h

```diff
@@ -142,13 +142,17 @@ static const INFORMATION_CLASS_INFO PsProcessInfoClass[] =
         ICIF_QUERY | ICIF_SET | ICIF_SET_SIZE_VARIABLE
     ),
 
-    /* ProcessUserModeIOPL */
+    /* ProcessUserModeIOPL is only implemented in x86 */
+#if defined (_X86_)
     IQS_SAME
     (
-        UCHAR,
+        0,
```

Comment:
This field is for the type and expands to `sizeof(Type)`, i.e. `sizeof(0)` == `sizeof(int), which is probably not what you want. I suggest keeping it as it is.

Reply:
The thing is, according to NT/2000 API reference documentation `ProcessUserModeIOPL` doesn't require a size length of any type hence it should be 0. Keeping UCHAR as it's is not correct either. Now that I think about it, are you OK with me having a new IQS macro specifically for these kind of particular classes like  ProcessUserModeIOPL, `IQS_NO_TYPE_LENGTH`, where the class has its own alginment but no type size length?

Reply:
Having a flag like you propose makes sense.